std/asys/native/filesystem/FileInfo.hx

```diff
@@ -0,0 +1,115 @@
+package asys.native.filesystem;
+
+import haxe.exceptions.NotImplementedException;
+import asys.native.system.SystemUser;
+import asys.native.system.SystemGroup;
+
+/**
+	Data from system call to `stat`.
+
+	TODO:
+	- Decide on data type for time fields: `Date` means additional allocations;
+		`Int` means "end-of-time" issue. Maybe `Float`?
+	- Decide on `ino` type: theoretically it could be any big number. For example
+		on Windows it could be a 64-bit unsigned integer. So may overflow.
+	- Decide on `size` type: `Int` limits `size` to ~2GB.
+**/
+private typedef NativeInfo = {
+	/** Time of last access (Unix timestamp) */
+	final atime:Int;
+	/** Time of last modification (Unix timestamp) */
+	final mtime:Int;
+	/** Time of last inode change (Unix timestamp) */
+	final ctime:Int;
+	/** Device number */
+	final dev:Int;
+	/** Owning user */
+	final uid:Int;
+	/** Owning group */
+	final gid:Int;
+	/** Inode number */
+	final ino:Int;
+	/** Inode protection mode */
+	final mode:Int;
+	/** Number of links */
+	final nlink:Int;
+	/** Device type, if inode device */
+	final rdev:Int;
+	/** Size in bytes */
+	final size:Int;
+	/** Block size of filesystem for IO operations */
+	final blksize:Int;
+	/** Number of 512-bytes blocks allocated */
+	final blocks:Int;
+};
+
+/**
+	Provides information about a file.
+**/
+@:coreApi
+abstract FileInfo(NativeInfo) from NativeInfo to NativeInfo {
```

Comment:
I see a problem here.
Just like in the other asys PR (and the current std), this (and other things) are a clone of the Unix api.
Which means that on Windows (and other non-unix operating systems) they will not work well.
A relevant blog post: <https://fasterthanli.me/articles/i-want-off-mr-golangs-wild-ride>
It's about Go, but many of the issues presented there apply to this PR (and Haxe's current standard library) too.